src/Maker/Security/AbstractSecurityMaker.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Maker\Security;
+
+use Doctrine\Bundle\DoctrineBundle\DoctrineBundle;
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\Exception\RuntimeCommandException;
+use Symfony\Bundle\MakerBundle\FileManager;
+use Symfony\Bundle\MakerBundle\Maker\AbstractMaker;
+use Symfony\Bundle\MakerBundle\Security\InteractiveSecurityHelper;
+use Symfony\Bundle\MakerBundle\Security\SecurityConfigUpdater;
+use Symfony\Bundle\MakerBundle\Security\SecurityControllerBuilder;
+use Symfony\Bundle\MakerBundle\Util\YamlSourceManipulator;
+use Symfony\Bundle\MakerBundle\Validator;
+use Symfony\Bundle\SecurityBundle\SecurityBundle;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ */
+abstract class AbstractSecurityMaker extends AbstractMaker
+{
+    protected const SECURITY_CONFIG_PATH = 'config/packages/security.yaml';
+
+    protected YamlSourceManipulator $ysm;
+    protected string $securityControllerName;
+    protected string $firewallToUpdate;
+    protected string $userClass;
+    protected string $userNameField;
+    protected bool $willLogout;
+
+    public function __construct(
+        protected FileManager $fileManager,
+        protected SecurityConfigUpdater $securityConfigUpdater,
+        protected SecurityControllerBuilder $securityControllerBuilder,
+    ) {
+    }
+
+    protected function runFixer(string $templateFilePath): void
+    {
+        $fixerPath = \dirname(__DIR__, 2).'/bin/php-cs-fixer-v3.13.0.phar';
+        $configPath = \dirname(__DIR__, 2).'/Resources/test/.php_cs.test';
```

Comment:
We're probably going to want to use the user's php cs config immediately, if it's available. For starters, we could look in the root dir for whatever the common filenames are. Then fallback to using our version.

Also, why not start doing this for *all* maker commands instead of security? It's a bit opinionated, but I think it's a good idea.

Finally, this would definitely be done better as an independent service, though I think I see why you avoided doing that: it's difficult to inject a service into *all* makers. It makes me wonder (even though I'm the king of "don't add extension points), if we might be better served dispatching an event at the end of running a maker - like right here - https://github.com/symfony/maker-bundle/blob/7c9e6968895c05c2757c08e63240eaba9e2b882e/src/Command/MakerCommand.php#L93 - then, we could ideally fetch whatever files were modified from the `Generator`. If those files end in `.php`, we php-cs-fix them.

Reply:
done in #1251